### PR TITLE
Bump Gemini models to 2.5

### LINKED ISSUE #5286: Update the fine-tuned Gemini model in tests to 2.5/3.0
We have "gemini-2.0-flash-lite-tuned" for tests. They're deprecating 2.0 in Q1, so we need to fine-tune 2.5 or 3.0 (latter if possible) and replace.

I migrated the non-tuned models in https://github.com/tensorzero/tensorzero/pull/5285

### PR DESCRIPTION
**Pull request by GabrielBianconi**
Fix #5286

 

---

> [!NOTE]
> **Upgrade Gemini models to 2.5 across the repo**
> 
> - Replace `gemini-2.0*`/`gemini-1.5*` with `gemini-2.5-*` in code, tests, configs, and examples (GCP Vertex and Google AI Studio), including shorthand `model` names and TOML blocks
> - Update GitHub Actions Vertex batch job to `model_name` `publishers/google/models/gemini-2.5-flash`
> - Adjust e2e/unit tests and fixtures: rename variants, update expected model IDs/URLs, add skips/conditions for 2.5 limitations (no penalties, tool/streaming edge cases), and tweak `max_tokens`/thinking parameters
> - Remove references to tuned flash-lite variants and migrate SFT recipes to `gemini-2.5-flash-lite`
> - Update parsing tests for shorthand URLs and example notebooks/scripts/constants
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 99963d8b3827f07fb70008f3c3835595c79f457d. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Review by shuyangli — CHANGES_REQUESTED** [review]
everything expect one spot looks good

File: docs/gateway/configuration-reference.mdx [hunk 1]
```diff
@@ -1287,10 +1287,10 @@ Defines the model ID of the GCP Vertex AI model.
 See [GCP Vertex AI's documentation](https://cloud.google.com/vertex-ai/generative-ai/docs/learn/model-versions) for the list of available model IDs.
 
 ```toml title="tensorzero.toml"
-[models."gemini-1.5-flash".providers.gcp_vertex]
+[models."gemini-2.5-flash".providers.gcp_vertex]
 # ...
 type = "gcp_vertex_gemini"
-model_id = "gemini-1.5-flash-001"
+model_id = "gemini-2.5-flash-001"
```
Comment (shuyangli) [inline docs/gateway/configuration-reference.mdx]: actually one more
